apps/web/components/utils.tsx

```diff
@@ -286,6 +310,26 @@ export async function getAllMCQs() {
   }
 }
 
+export async function updateUserHistory(email: string, trackId: string, problemId: string) {
+  try {
+    const history = await db.history.upsert({
+      where: { id: email + trackId },
+      update: { problemId: problemId, updatedAt: new Date(), userEmail: email },
+      create: {
+        id: email + trackId,
```

Comment:
why are you creating an id yourself?

Reply:
because it will create a unique identifier for the user with the track we dont have to take email and trackid seaperatly it will benefit us in long run.

Reply:
But you are still storing user email and trackId separately.

Reply:
its foriegn key not its primary key, there should be a unique key to it .

Reply:
https://www.prisma.io/docs/orm/prisma-schema/data-model/relations/relation-mode#:~:text=If%20you%20use%20Prisma%20ORM,primary%20key%20in%20another%20table.

Reply:
what do you want to say? the email and tracks in history Table is foreign keys but there is no unique key history table.
its id is its unique key .
ok tell me if in future you want to take a single row from history and connect it to a another table how you will do that without using history table id .

Reply:
Why are you not letting the db handle history table unique id like for othe tables?

Reply:
because it does not matter + it makes the id meaningfull

Reply:
https://stackoverflow.com/a/59401578
Its best to use whatever is provided by the database.

Reply:
you must be focusing on 2 and 3 points of the para so for 2nd its personal preference + it making id more readable for 3rd you are sending 2 input email and traceid where as i am sending 1 , yeah i know the combine length of data is same so the compressed size must be comming same but sending two key value pair is little expensive computation than 1 
 
for 1st point i dont think we have to take daedlock and delay into account as its happen very rarely + the database take care of it for us so dont worry.